### PR TITLE
Browser File IO - add Apollo Client and relate-scripts component in sidebar

### PR DESCRIPTION
**Pull request by jk05**
- **NO NEED TO PR AT THE MOMENT. HAVE TO MAKE SEVERAL LARGE CHANGES IN A SEPARATE BRANCH DUE TO CHANGING REQUIREMENTS THAT I WILL MERGE IN AND THEN UPDATE THIS PR**
- I may use this as my main feature branch and just merge into this going forward
- Add in Apollo Client and Apollo Upload Client
- Created a Relate Scripts Component to display in the Sidebar, displaying all the Project Files
- Had to proxy requests to relate-web for now as getting CORS issues when trying to use Relate API url passed from Desktop. Will work on this in an upcoming PR

Small adjustments to the saved-scripts component here: https://github.com/neo4j-apps/relate-by-ui/pull/7, which are visible in the screenshot. Remove export/new folder and editing folder/file names in the sidebar

### CODE REVIEW

File: build_scripts/webpack.config.js [hunk 1]
```diff
@@ -103,6 +103,10 @@ module.exports = {
     host: '0.0.0.0',
     port: 8080,
     disableHostCheck: true,
-    hot: !helpers.isProduction
+    hot: !helpers.isProduction,
+    proxy: {
```
Comment (jk05) [inline build_scripts/webpack.config.js:107]: proxy for now until we use params passed from Desktop and a potential CORS issue that was being experienced

File: src/browser/modules/Sidebar/relate-scripts.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  return MyScripts({
+    ...props,
+    scripts,
+    isRelateScripts: true,
+    onRemoveScript: favorite => {
+      const directory = favorite.path.substring(1) // @todo: adding in SLASH to path
+      const filePath = path.join(directory, favorite.name)
+      return removeFavorite({
+        variables: removeProjectFileMutationVars(filePath),
+        update: (cache, { data: { removeProjectFile } }) => {
```
Comment (jk05) [inline src/browser/modules/Sidebar/relate-scripts.js:101]: update cache

File: src/browser/modules/Sidebar/relate-scripts.utils.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+})
+
+export const mapRelateFavorites = async ({
+  downloadToken,
+  name,
+  directory
+}) => ({
+  id: uuid.v4(),
+  name,
+  path: directory.startsWith('.') ? SLASH : `${SLASH}${directory}`, // @todo: need to look into this
```
Comment (jk05) [inline src/browser/modules/Sidebar/relate-scripts.utils.js:26]: couldnt get this working with `.` which is the root returned from Relate. Had to use `SLASH` for now just to get it to show, will look into later as its not a blocking issue

File: src/browser/modules/Sidebar/relate-scripts.utils.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  contents: await getProjectCypherFileContents(downloadToken, name)
+})
+
+const getProjectCypherFileContents = (token, name) =>
+  remote
+    .request('GET', `/files/${token}/${name}`)
+    .then(body => body.text())
+    .catch(e => console.log('++err', e)) // ?
```
Comment (jk05) [inline src/browser/modules/Sidebar/relate-scripts.utils.js:34]: need to look into handling errors and what to do in next few PRs

Comment (jk05) [reply src/browser/modules/Sidebar/relate-scripts.utils.js:34]: Any suggestions here are welcome

File: src/browser/modules/Sidebar/relate-scripts.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    removeProjectFile(name: $projectId, filePath: $filePath) {
+      name
+      directory
+    }
+  }
+`
+
+const MyScriptsComponent = props => {
+  // @todo: handling loading and error?? Pass to MyScripts??
+  const { loading, error, data, refetch } = useQuery(GET_PROJECT_FILES, {
```
Comment (jk05) [inline src/browser/modules/Sidebar/relate-scripts.js:59]: error or loading handling suggestions welcome here

File: src/browser/modules/Sidebar/relate-scripts.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+  return MyScripts({
+    ...props,
+    scripts,
+    isRelateScripts: true,
+    onRemoveScript: favorite => {
+      const directory = favorite.path.substring(1) // @todo: adding in SLASH to path
+      const filePath = path.join(directory, favorite.name)
+      return removeFavorite({
+        variables: removeProjectFileMutationVars(filePath),
```
Comment (jk05) [inline src/browser/modules/Sidebar/relate-scripts.js:100]: need to think of error handling here too

File: src/browser/modules/Sidebar/relate-scripts.js (hunk 5 above)
Comment (OskarDamkjaer) [reply src/browser/modules/Sidebar/relate-scripts.js:59]: What can cause errors? Does it only error if the connection is broken? Just thinking how serious the errors are

File: src/browser/modules/Sidebar/relate-scripts.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  const { loading, error, data, refetch } = useQuery(GET_PROJECT_FILES, {
+    variables: getProjectFilesQueryVars
+  })
+  const [removeFavorite] = useMutation(DELETE_PROJECT_FILE)
+
+  const [scripts, setScripts] = useState([])
+  const isMountedRef = useRef(false)
+
+  useEffect(() => {
+    isMountedRef.current = true
```
Comment (OskarDamkjaer) [inline src/browser/modules/Sidebar/relate-scripts.js:68]: This doesn't need to be a useRef, a closure is enough. I changed it in my pr :)

File: src/browser/modules/Sidebar/relate-scripts.js (hunk 5 above)
Comment (jk05) [reply src/browser/modules/Sidebar/relate-scripts.js:59]: network error like when connection is broken I suppose. Or errors from the graphQL endpoint

Comment (OskarDamkjaer) [reply src/browser/modules/Sidebar/relate-scripts.js]: Hm.. perhaps it's worth having an errorbanner like db not available but for network? For errors from the end point I'm not sure.. We want them in sentry eventually at least

Comment (jk05) [reply src/browser/modules/Sidebar/relate-scripts.js]: yeah the banner makes sense. I will look into that, I didnt look in to the error banners

**Comment by jk05** [comment]
closed in favour of https://github.com/neo4j/neo4j-browser/pull/1209

**Pull request closed without merging**